src/rpcmining.cpp

```diff
@@ -45,9 +45,13 @@ Value GetNetworkHashPS(int lookup, int height) {
     if (pb == NULL || !pb->nHeight)
         return 0;
 
-    // If lookup is -1, then use blocks since last difficulty change.
+    // If lookup is -1, then use difficulty averaging window.
     if (lookup <= 0)
-        lookup = pb->nHeight % Params().GetConsensus().DifficultyAdjustmentInterval() + 1;
+        lookup = pb->nHeight - Params().GetConsensus().nPowAveragingWindow;
+
+    // If lookup is still negative, then use blocks since genesis.
+    if (lookup <= 0)
+        lookup = pb->nHeight;
```

Comment:
No, because `lookup` is used as a counter, not an index into the blockchain. I'll merge this `if` with the existing one below it to make this clearer.